__tests__/app.test.js

```diff
@@ -55,6 +55,57 @@ describe("GET /api/articles", () => {
         });
       });
   });
+  test("returns 200 and relevant article when queried", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch&sort_by=article_id&order=asc")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeSortedBy("article_id")
+        expect(articles).toHaveLength(11);
+        articles.forEach((article) => {
+          expect(article).toEqual(
+            expect.objectContaining({
+              author: expect.any(String),
+              title: expect.any(String),
+              article_id: expect.any(Number),
+              topic: "mitch",
+              created_at: expect.any(String),
+              votes: expect.any(Number),
+              comment_count: expect.any(Number),
+            })
+          );
+        });
+      });
+  });
+  test("defaults to descending date when only topic query given", () => {
+    return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then(({ body: { articles } }) => {
+        expect(articles).toBeSortedBy("created_at", {
+          descending: true,
+          coerce: true,
+        });
+      });
+  });
+    test("returns 400 for invalid query", () => {
+      return request(app)
+        .get("/api/articles?topic=mitch&sort_by=temperament&order=sideways")
+        .expect(400)
+        .then(({ body: { msg } }) => {
+          expect(msg).toBe('bad request')
+
+        });
+    });
+      test("returns [] for topic does not exist", () => {
+      return request(app)
+        .get("/api/articles?topic=emus")
+        .expect(200)
+        .then(({ body: { articles } }) => {
+          expect(articles).toEqual([])
+
+        });
+    });
```

Comment:
We should add the following test cases:
- sort_by =banana
- topic is valid but does not have an article i.e. paper and responds with an empty array status 200